packages/plugins/typescript/typescript/src/visitor.ts

```diff
@@ -30,6 +31,8 @@ export interface TypeScriptPluginParsedConfig extends ParsedTypesConfig {
   constEnums: boolean;
   enumsAsTypes: boolean;
   futureProofEnums: boolean;
+  futureProofUnions: boolean;
+  nonOptionalTypename: boolean;
```

Comment:
Can you please explain why `nonOptionalTypename` is needed?

Reply:
removed!